ci/pipeline/determine_tests_to_run.py

```diff
@@ -32,58 +33,44 @@ def list_changed_files(commit_range):
     return [s.strip() for s in out.decode().splitlines() if s is not None]
 
 
-def is_pull_request():
+def _is_pull_request():
     return os.environ.get("BUILDKITE_PULL_REQUEST", "false") != "false"
 
 
-def get_commit_range():
+def _get_commit_range():
     return "origin/{}...{}".format(
         os.environ["BUILDKITE_PULL_REQUEST_BASE_BRANCH"],
         os.environ["BUILDKITE_COMMIT"],
     )
 
 
+_ALL_TAGS = set(
+    " ".join(
+        [
+            "lint python cpp core_cpp java workflow accelerated_dag dashboard",
+            "data serve ml tune train llm rllib rllib_gpu rllib_directly",
+            "linux_wheels macos_wheels docker doc python_dependencies tools",
+            "release_tests compiled_python",
+        ]
+    ).split()
+)
+
 if __name__ == "__main__":
     assert os.environ.get("BUILDKITE")
 
     parser = argparse.ArgumentParser()
     args = parser.parse_args()
 
-    RAY_CI_ML_AFFECTED = 0
-    RAY_CI_TUNE_AFFECTED = 0
-    RAY_CI_TRAIN_AFFECTED = 0
-    RAY_CI_LLM_AFFECTED = 0
-    # Whether only the most important (high-level) RLlib tests should be run.
-    # Set to 1 for any changes to Ray Tune or python source files that are
-    # NOT related to Serve, Dashboard or Train.
-    RAY_CI_RLLIB_AFFECTED = 0
-    RAY_CI_RLLIB_GPU_AFFECTED = 0
-    # Whether all RLlib tests should be run.
-    # Set to 1 only when a source file in `ray/rllib` has been changed.
-    RAY_CI_RLLIB_DIRECTLY_AFFECTED = 0
-    RAY_CI_SERVE_AFFECTED = 0
-    RAY_CI_CORE_CPP_AFFECTED = 0
-    RAY_CI_CPP_AFFECTED = 0
-    RAY_CI_JAVA_AFFECTED = 0
-    RAY_CI_PYTHON_AFFECTED = 0
-    RAY_CI_LINUX_WHEELS_AFFECTED = 0
-    RAY_CI_MACOS_WHEELS_AFFECTED = 0
-    RAY_CI_DASHBOARD_AFFECTED = 0
-    RAY_CI_DOCKER_AFFECTED = 0
-    RAY_CI_DOC_AFFECTED = 0
-    RAY_CI_PYTHON_DEPENDENCIES_AFFECTED = 0
-    RAY_CI_TOOLS_AFFECTED = 0
-    RAY_CI_DATA_AFFECTED = 0
-    RAY_CI_WORKFLOW_AFFECTED = 0
-    RAY_CI_RELEASE_TESTS_AFFECTED = 0
-    RAY_CI_COMPILED_PYTHON_AFFECTED = 0
-    RAY_CI_ACCELERATED_DAG_AFFECTED = 0
-    # Whether to run jobs that are affected by any changes
-    RAY_CI_LINT_AFFECTED = 1
+    tags: Set[str] = set()
+
+    tags.add("lint")
 
-    if is_pull_request():
-        commit_range = get_commit_range()
-        files = list_changed_files(commit_range)
+    def x(line: str):
```

Comment:
```suggestion
    def _update_tags(line: str):
```
(if that's accurate)

Reply:
hm why not take variadic arguments or a list instead of splitting the line?

Reply:
> (if that's accurate)

changed to `_emit`